src/python_testing/TC_SU_2_7.py

```diff
@@ -0,0 +1,466 @@
+        state_transition_event_handler = EventSubscriptionHandler(
+            expected_cluster=self.ota_req, expected_event_id=self.ota_req.Events.StateTransition.event_id)
+        await state_transition_event_handler.start(controller, self.requestor_node_id, endpoint=0, min_interval_sec=0, max_interval_sec=max_interval)
+        await self.announce_ota_provider(controller, self.provider_node_id, self.requestor_node_id)
+        # Register event, should change to Querying
+        event_report = state_transition_event_handler.wait_for_event_report(self.ota_req.Events.StateTransition, timeout_sec=60)
+        logger.info(f"Event report {event_report}")
+        self.verify_state_transition_event(event_report, self.ota_req.Enums.UpdateStateEnum.kIdle,
+                                           self.ota_req.Enums.UpdateStateEnum.kQuerying, expected_target_version=NullValue)
+
+        # Event for Downloading
+        event_report = state_transition_event_handler.wait_for_event_report(self.ota_req.Events.StateTransition, timeout_sec=60)
+        logger.info(f"Event report for Downloading {event_report}")
+        self.verify_state_transition_event(event_report, self.ota_req.Enums.UpdateStateEnum.kQuerying,
+                                           self.ota_req.Enums.UpdateStateEnum.kDownloading, expected_target_version=self.expected_software_version)
+
+        # Event for Applying
+        event_report = state_transition_event_handler.wait_for_event_report(
+            self.ota_req.Events.StateTransition, timeout_sec=self.ota_image_download_timeout)
+        logger.info(f"Event report for Applying {event_report}")
+        self.verify_state_transition_event(event_report, self.ota_req.Enums.UpdateStateEnum.kDownloading,
+                                           self.ota_req.Enums.UpdateStateEnum.kApplying, expected_target_version=self.expected_software_version)
+        state_transition_event_handler.cancel()
+        Clusters.OtaSoftwareUpdateRequestor.Attributes.FeatureMap
+        # kIdle ( After update is Optional) in this case we only wait state to be Idle
+        # Just wait the device to be kIdle to avoid unexpected states in following Steps.
+        update_state_match = AttributeMatcher.from_callable(
+            "Update state is kIdle",
+            lambda report: report.value == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kIdle)
+        update_state_attr_handler.await_all_expected_report_matches([update_state_match], timeout_sec=600)
+        # Added this block because after AttributeSubscriptionHandler
+        # and calling an await method it was triggering an TimeoutError and CancelledError
+        # this is just to catch it and continue with the test
+        update_state_attr_handler.cancel()
+        with contextlib.suppress(asyncio.CancelledError):
+            await asyncio.sleep(0.1)
+
+        # Verify VersionAppliedEvent
```

Comment:
On a normal device, is this generated after restart? If so, you'll need to verify that the device is back from reset before attempting to read this or you're going to get a read error when the device can't be found.